presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -20,14 +20,16 @@ namespace facebook::presto::http {
 HttpClient::HttpClient(
     folly::EventBase* eventBase,
     const folly::SocketAddress& address,
-    std::chrono::milliseconds timeout)
+    std::chrono::milliseconds timeout,
+    const std::function<void(int)> reportStatsFunc)
```

Comment:
Either
```
const std::function<void(int)>& reportStatsFunc
reportStatsFunc_(reportStatsFunc)
```
or
```
std::function<void(int)> reportStatsFunc
reportStatsFunc_(std::move(reportStatsFunc))
```
I think the latter is better as you construct report stats lamda function on-demand? Thanks!

Reply:
So why not just std::function<void(int)>&& reportStatsFunc for a better indication of on-demand creation?